src/OrchardCore.Modules/OrchardCore.Settings/Razor/SiteRazorHelperExtensions.cs

```diff
@@ -0,0 +1,20 @@
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using OrchardCore;
+using OrchardCore.Settings;
+
+public static class SiteRazorHelperExtensions
+{
+    /// <summary>
+    /// Gets the site name.
+    /// </summary>
+    /// <param name="orchardHelper">The <see cref="IOrchardHelper"/>.</param>
+    public static async Task<string> GetSiteName(this IOrchardHelper orchardHelper)
+    {
+        var siteService = orchardHelper.HttpContext.RequestServices.GetService<ISiteService>();
```

Comment:
The rest of the code expects that this service will be found, so this should probably be `GetRequiredService<ISiteService>()`

Reply:
I think we use `GetService()` in other `IOrchardHelper` extensions

Reply:
@hishamco 
If you use `GetService<>` you should check if the resolved value is not null before you use it's services. But why do you need this extension? if you want to `Site` info in the views, you can just access the `Site` object in your views. `Site` is available when `@inherits OrchardCore.DisplayManagement.Razor.RazorPage<TModel>` is used.

Reply:
> If you use GetService<> you should check if the resolved value is not null before you use it's services

I knew what you mean, I just followed what we did before, which might need to be fixed

> if you want to Site info in the views, you can just access the Site object in your views. 

The intention for `IOrchardHelper` to create a shortcut for many of the useful things that user need to access with simple dot accessor for the helper

Reply:
One can still using `Site` from the views. I don't think this extension is useful.

Reply:
I never mind closing this one, but the user should know in advance what is available in the view, again what I saw in many CMSs that the helper is convenient for enormous things. @sebastienros do you still think there's value in adding this, or the link in the issue description before adding `Site` property on the view?

Reply:
@hishamco the Site property already exists. The use don't need to add anything to use it

Reply:
Do you want to add some docs instead so one can be more aware of the existing property, or should we close?

Reply:
I don't think there's an issue with the extension method but if you all disagree let us close it

Reply:
If you look at the comment https://github.com/OrchardCMS/OrchardCore/issues/3553#issuecomment-488806086 you will see such API is useful, especially in decoupled CMS

I submitted this PR at the time because I'm heavily using decoupled CMS

Reply:
I've read that comment. However, apparently, neither Dody nor Sebastien knew about the `Site` property.